Upgrade to sandbox version 3 on iOS and macOS

pvollan opened the pull request:
#### 129d7629be7ded74ee0c924c96c48c1d0372d61d
<pre>
Upgrade to sandbox version 3 on iOS and macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=248399">https://bugs.webkit.org/show_bug.cgi?id=248399</a>
rdar://102719029

Reviewed by Brent Fulgham.

Upgrade to sandbox version 3 on iOS and macOS. This patch should not introduce any behavior change, since
the resources with implicit access in version 1 has been explicitly allowed. Also, this patch adds more
sandbox include files to be able to share common rules.

* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Shared/Sandbox/iOS/common.sb
* Source/WebKit/Shared/Sandbox/macOS/common.sb:
* Source/WebKit/Shared/Sandbox/util.sb
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/257272@main">https://commits.webkit.org/257272@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/509c22f9a770663b6af27f95b07219d78fcecc60

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98257 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7467 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31395 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107710 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7968 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/36228 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/90840 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/104292 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103911 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5988 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84850 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33087 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87870 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89557 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76031 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1427 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20989 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1380 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22492 "Found 1 new test failure: streams/readable-stream-default-controller-error.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6264 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44996 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2502 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2721 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41919 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WTF/wtf/PlatformUse.h:
```diff
@@ -375,3 +375,8 @@
 #endif
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000)
```
I am always confused by when to use `PLATFORM(IOS_FAMILY)` versus `PLATFORM(IOS)`. I believe that this rule means that USE_SANDBOX_VERSION_3 will be undefined for watchOS and tvOS. Is this your intention?

pvollan replied:
> I am always confused by when to use `PLATFORM(IOS_FAMILY)` versus `PLATFORM(IOS)`. I believe that this rule means that USE_SANDBOX_VERSION_3 will be undefined for watchOS and tvOS. Is this your intention?

That is a good point. The intention is for also watchOS and tvOS to have version 3. I have removed this from the latest patch, and will enable in a separate patch.

Thanks for reviewing!

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
```diff
@@ -21,50 +21,24 @@
 ;;;
 ;;; The following rules were originally contained in 'common.sb'. We are duplicating them here so we can
 ;;; remove unneeded sandbox extensions.
 ;;;
```
Perhaps this comment should move to `Shared/Sandbox/iOS/common.sb`, too?

pvollan replied:
There are still quite a few rules that were not moved to Shared/Sandbox/iOS/common.sb in this patch, so I think we may leave the comment for now. However, we should move all the common rules to the new file including the comment. I will address that in a separate patch.

Thanks for reviewing!

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
```diff
@@ -21,46 +21,50 @@
 ;;;
 ;;; The following rules were originally contained in 'common.sb'. We are duplicating them here so we can
 ;;; remove unneeded sandbox extensions.
 ;;;
```
Ditto: Perhaps move the comment to `Shared/Sandbox/iOS/common.sb` as well?

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -21,78 +21,19 @@
 ;;;
 ;;; The following rules were originally contained in 'common.sb'. We are duplicating them here so we can
 ;;; remove unneeded sandbox extensions.
 ;;;
```
Ditto: move comment to `Shared/Sandbox/iOS/common.sb`.

brentfulgham reviewed (approved):
A very nice cleanup! r=me

pvollan committed:
Upgrade to sandbox version 3 on iOS and macOS
https://bugs.webkit.org/show_bug.cgi?id=248399
rdar://102719029

Reviewed by Brent Fulgham.

Upgrade to sandbox version 3 on iOS and macOS. This patch should not introduce any behavior change, since
the resources with implicit access in version 1 has been explicitly allowed. Also, this patch adds more
sandbox include files to be able to share common rules.

* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Shared/Sandbox/iOS/common.sb
* Source/WebKit/Shared/Sandbox/macOS/common.sb:
* Source/WebKit/Shared/Sandbox/util.sb
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/257272@main
